src/query.rs

```diff
@@ -0,0 +1,51 @@
+use std::any::type_name;
+
+use crate::{authentication::authenticate_user, order::Order};
+use async_graphql::{Context, Error, Object, Result};
+
+use bson::Uuid;
+use mongodb::{bson::doc, Collection, Database};
+use serde::Deserialize;
+
+/// Describes GraphQL invoice queries.
+pub struct Query;
+
+#[Object]
+impl Query {
```

Comment:
where is the invoice entity resolver?
also a employee/admin query to get all invoices might be helpful

Reply:
Is it possible to add this to the subgraph?
CI:
```
 [service] On type "Order", for @key(fields: "id invoice { id }", resolvable: false): Cannot query field "id" on type "Invoice" (the field should either be added to this subgraph or, if it should not be resolved by this subgraph, you need to add it to this subgraph with @external).
 ```

Reply:
payments also need to refer to the invoice:
![image](https://github.com/MiSArch/invoice/assets/53957498/e8422123-4dee-4d76-a91e-0139f61ac80d)
and I would guess that the payment service should handle this relation

Reply:
The invoice should be able to be queried before the payment is created, what if the payment fails or cannot be created? Then the invoice should still be available

Reply:
yes?
I just want an entity resolver to be available, that's it 😅 
of course invoice should still be responsible for resolving invoices based on the relation to Order
payment should just be additionally capable of relating to invoices

Reply:
I have added the invoice entity resolver as well: [5195b99](https://github.com/MiSArch/invoice/pull/1/commits/5195b9901d966d5e7b8f899c57f8d27787ac3b80)